unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/AssetBundles/AB/Providers/AssetBundleWebLoader.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Cysharp.Threading.Tasks;
+using MainScripts.DCL.Controllers.AssetManager;
+using System.Threading;
+using UnityEngine;
+
+namespace DCL.Providers
+{
+    public class AssetBundleWebLoader : AssetBundleWebRequestBasedProvider, IAssetBundleProvider
+    {
+        private const string CACHING_FEATURE_FLAG = "ab_caching";
+
+        private Service<IWebRequestController> webRequestController;
+        private readonly DataStore_FeatureFlag featureFlags;
+        private readonly DataStore_Performance performance;
+
+        private static int maxConcurrentRequests => CommonScriptableObjects.rendererState.Get() ? 30 : 256;
+        private bool cachingEnabled => featureFlags.flags.Get().IsFeatureEnabled(CACHING_FEATURE_FLAG);
+
+        public AssetBundleWebLoader(DataStore_FeatureFlag featureFlags, DataStore_Performance performance)
+        {
+            this.featureFlags = featureFlags;
+            this.performance = performance;
+        }
+
+        public async UniTask<AssetBundle> GetAssetBundleAsync(string contentUrl, string hash, CancellationToken cancellationToken)
+        {
+            await WaitForConcurrentRequestsSlot();
+            performance.concurrentABRequests.Set(performance.concurrentABRequests.Get() + 1);
+
+            try
+            {
+                var url = contentUrl + hash;
+                var hash128 = Hash128.Compute(hash);
```

Comment:
Unless I'm mistaken, this hash is the `asset` hash. That means it wont change even if the asset is reuploaded with a new version but we would still load an old version.

Reply:
According to [this ADR](https://adr.decentraland.org/adr/ADR-11) I will need to include "version" in the hash

Reply:
make sense, thx!

Reply:
Got it covered with respect of "vXXX" of the endpoint, apart of that the hash is unique